jgrapht-core/src/main/java/org/jgrapht/graph/specifics/FastLookupDirectedSpecifics.java

```diff
@@ -105,11 +133,12 @@ public FastLookupDirectedSpecifics(
     @Override
     public E getEdge(V sourceVertex, V targetVertex)
     {
-        List<E> edges = touchingVerticesToEdgeMap.get(new Pair<>(sourceVertex, targetVertex));
+        Set<E> edges = touchingVerticesToEdgeMap.get(new Pair<>(sourceVertex, targetVertex));
         if (edges == null || edges.isEmpty())
             return null;
-        else
-            return edges.get(0);
+        else { 
+            return edges.stream().findFirst().orElse(null);
```

Comment:
Is there an advantage to this over edges.iterator().next()?  There's no need for the option check since we already tested for isEmpty above.